src/index.js

```diff
@@ -488,9 +493,37 @@ class EventsCoreAPI {
    * @param {string} signatureHeaderValue Value of x-adobe-signature header in each POST request to the registered webhook URL
    * @returns {boolean} If signature matches return true else return false
    */
-  verifySignatureForEvent (event, clientSecret, signatureHeaderValue) {
-    const hmacDigest = Base64.stringify(hmacSHA256(JSON.stringify(event), clientSecret))
-    return hmacDigest === signatureHeaderValue
+  /**
+   * Authenticating events by verifying signature
+   * 
+   * @param {*} event JSON payload delivered to the registered webhook URL
+   * @param {*} clientSecret Client secret can be retrieved from the Adobe I/O Console integration
+   * @param {*} recipientClientId Target recipient client id retrieved from the Adobe I/O Console integration
+   * @param {*} deprecatedSignature Value of HMAC signature retrieved from the x-adobe-signature header in each POST request to the registered webhook URL
+   * @param {*} digiSignature1 Value of digital signature retrieved from the x-adobe-digital-signature1 header in each POST request to webhook
+   * @param {*} digiSignature2 Value of digital signature retrieved from the x-adobe-digital-signature2 header in each POST request to webhook
+   * @param {*} publicKeyUrl1 Value of public key url retrieved from the x-adobe-public-key1-url header in each POST request to webhook
+   * @param {*} publicKeyUrl2 Value of public key url retrieved from the x-adobe-public-key2-url header in each POST request to webhook
+   * @returns {boolean} If signature matches return true else return false
+   */
+  async verifySignatureForEvent (event, clientSecret, recipientClientId, deprecatedSignature, digiSignature1, digiSignature2, publicKeyUrl1, publicKeyUrl2) {
```

Comment:
I am confused. This is an SDK, so the new method should not be taking in `clienSecret` as a parameter. It should also not implement any fallback verification strategy.
We shoud instead mark the old method as deprecated and open a new method for verifying signatures using digital signature.

Reply:
Wouldn't it be simpler for the client if he can simply pass in the request headers and we do the extraction of the digital signature?
We can have 2 methods exposed, one with request headers and one that accepts the required params?
Also, instead of accepting all digital signature params in a flattened form, it would be better if can accept a concrete object (or atleast a map):
```javascript
signatureOptions: { 
  digiSignature1: '',
  digiSignature2: '',
  publicKeyUrl1: '',
  publicKeyUrl2: ''
 }
```

Reply:
>This is an SDK, so the new method should not be taking in clientSecret as a parameter. It should also not implement any fallback verification strategy.
`client_secret` was there earlier too. not sure, if that is the case as you described. As user would be having their secret with them they can use it to call the sdk method. would like @sangeetha5491 's opinion here.

Reply:
reasonable
>We shoud instead mark the old method as deprecated and open a new method for verifying signatures using digital signature.

Reply:
They are not getting used as `headers` per say within the api, so don't think this seems important to have 2 methods exposed. However, to make the api simpler we can batch the signature params in a map or so as you mentioned.

Reply:
> > This is an SDK, so the new method should not be taking in clientSecret as a parameter. It should also not implement any fallback verification strategy.
> 
> `client_secret` was there earlier too. not sure, if that is the case as you described. As user would be having their secret with them they can use it to call the sdk method. would like @sangeetha5491 's opinion here.

I understand that but what I am saying is that there is no need to pass the `client_secret` in the new method of signature verification. For the client, there would be two methods essentially:

1. Verify signature using client_secret (marked deprecated)
2. Verify signature using digital signature. We don't need the `client_secret` here

Reply:
Got that. have done the refactoring accordingly.